### ISSUE TITLE
Atlas tissue probability map for 

### ISSUE THREAD

**Issue by m-dannecker** [neel-dey/Atlas-GAN#9]
Hey Neel,

Thank you for making the code-base to your paper public. I'm planning to run Atlas-GAN on dhcp-neonates and was wondering how to generate the label maps that you reported in the paper. In the paper it says they are "generated by majority voting on inverse warped labels of training images". If I understand correctly, you use the learned inverse deformation of each training-subject to warp the training-subject label maps to the template and thus generate the tissue-probability map by label voting. Is there any code available for this? Or what would you recommend is the easiest way to, during test-time, not only get the atlas intensities but also the tissue-maps, ideally from the "GT" segmentations of the data? 
Thanks in advance for your help!

Best,

Maik

**Comment by neel-dey** [comment]
Hi @m-dannecker ,

Thanks for your interest! The text you're referring to is how we handled the Huntington's dataset. 

For conditional dHCP templates, we just used [DrawEM](https://github.com/MIRTK/DrawEM) to segment the generated intensity templates (please see Appendix B.1.2). We assumed no access to image segmentations at training time and these segmentations were only used for evaluation, not for constructing the atlases. 

Using segmentation supervision during training does generally improve the atlas quality though and if you want to incorporate that within our framework: just edit the generator network's output to also produce an additional N-channel segmentation volume with a softmax activation. Then train the registration network in the same way as a supervised voxelmorph network by adding a Dice loss. The discriminator input/output can stay the same.

Let me know if you have any other questions :)

P.S. Just FYI, make sure to affinely prealign all the volumes first for dHCP. Rigid-only prealignment also works with our framework (see below for a quick snapshot of a random old experiment), but it'll take longer to train and I suggest doubling `g_ch` (the channel width multiplier for the generator network) to 64 in that case.

![image](https://github.com/user-attachments/assets/4fbf04a9-f56a-4d01-8901-b411ecbe29d1)

**Comment by m-dannecker** [comment]
Hey Neel,

thanks for the fast reply! That helps me already. 

Best,

Maik
